packages/onboarding/src/cart/index.tsx

```diff
@@ -0,0 +1,317 @@
+	productsList: Record< string, ProductListItem >,
+	domainItem?: CartItem
+) => {
+	const siteUrl = domainItem?.meta;
+
+	const newCartItems = [ domainItem ].filter( ( item ) => item );
+	const isFreeThemePreselected = startsWith( themeSlugWithRepo, 'pub' );
+	// x	const bearerToken = get( getSignupDependencyStore( state ), 'bearer_token', null );
+
+	if ( isManageSiteFlow ) {
+		const siteSlugManaged = getSignupCompleteSlug();
+		const newCartItems = [ domainItem ].filter( ( item ) => item );
+
+		await processItemCart(
+			newCartItems,
+			siteSlugManaged,
+			isFreeThemePreselected,
+			themeSlugWithRepo,
+			flowName,
+			userIsLoggedIn,
+			productsList
+		);
+		return;
+	}
+
+	const newSiteParams = getNewSiteParams( {
+		flowToCheck: flowName,
+		isPurchasingDomainItem,
+		themeSlugWithRepo,
+		siteUrl,
+		siteTitle,
+		siteAccentColor,
+		useThemeHeadstart,
+		comingSoon,
+	} );
+
+	// if ( isEmpty( bearerToken ) && 'onboarding-registrationless' === flowToCheck ) {
+	// 	saveToLocalStorageAndProceed( state, domainItem, themeItem, newSiteParams, callback );
+	// 	return;
+	// }
+
+	const locale = getLocaleSlug();
+
+	const siteCreationResponse: NewSiteSuccessResponse = await wpcom.req.post( '/sites/new', {
+		...newSiteParams,
+		locale,
+		lang_id: getLanguage( locale as string )?.value,
+		client_id: config( 'wpcom_signup_id' ),
+		client_secret: config( 'wpcom_signup_key' ),
+	} );
+
+	if ( ! siteCreationResponse.success ) {
+		// TODO ebuccelli: Manage siteCreationResponse.errors
+		return;
+	}
+
+	const parsedBlogURL = getUrlParts( siteCreationResponse?.blog_details.url );
+	const siteSlug = parsedBlogURL.hostname;
+	const siteId = siteCreationResponse?.blog_details.blogid;
+	const providedDependencies = {
+		siteId,
+		siteSlug,
+		domainItem,
+	};
+
+	if ( isTailoredSignupFlow( flowName ) ) {
+		await setupSiteAfterCreation( { siteId, flowName: flowName } );
+	}
+
+	await processItemCart(
+		newCartItems,
+		siteSlug,
+		isFreeThemePreselected,
+		themeSlugWithRepo,
+		flowName,
+		userIsLoggedIn,
+		productsList
+	);
+
+	return providedDependencies;
+};
+
+function prepareItemForAddingToCart( item: CartItem, lastKnownFlow?: string ) {
+	return {
+		...item,
+		extra: {
+			...item.extra,
+			context: 'signup',
+			...( lastKnownFlow && { signup_flow: lastKnownFlow } ),
+		},
+	};
+}
+
+export async function addPlanToCart(
+	siteSlug: string,
+	flowName: string,
+	userIsLoggedIn: boolean,
+	themeSlugWithRepo: string,
+	productsList: Record< string, ProductListItem >,
+	cartItem: CartItem
+) {
+	if ( isEmpty( cartItem ) ) {
+		// the user selected the free plan
+		return;
+	}
+
+	const isFreeThemePreselected = startsWith( themeSlugWithRepo, 'pub' );
+	const newCartItems = [ cartItem ].filter( ( item ) => item );
+
+	await processItemCart(
+		newCartItems,
+		siteSlug,
+		isFreeThemePreselected,
+		themeSlugWithRepo,
+		flowName,
+		userIsLoggedIn,
+		productsList
+	);
+}
+
+const addPrivacyProtectionIfSupported = (
```

Comment:
I may be wrong (I bow to the domain team @Automattic/nomado ) but I think that this function has no effect and could be removed. See https://github.com/Automattic/wp-calypso/pull/63292 where this behavior was removed from checkout.

Reply:
Thanks for suggesting this, would be nice not to have to load all the `productsList`.
I see that this function adds `extra: {privacy: true}` to the cartItem when called, but I don't know if that is useful.
I hope @Automattic/nomado can give us some more insight :)

Reply:
re: Privacy protection, I found a conversation I had with Nomado about it in Slack: p1651677667181089-slack-C0BNMNMNG

Reply:
That's so great to hear @sirbrillig, I'll remove the function then. Thanks a lot.